src/topmodel.c

```diff
@@ -508,9 +508,189 @@ if(yes_print_output==TRUE)
 return;
 }
 
+
+/*********************************
+ * Define functions to be used in init() and in Set_value in bmi_topmodel.c
+ * to enable calibratable parameters to be updated
+ * **********************************/
+
+
+// FUNCTION TO CONVERT DISTANCE/AREA FORM TO TIME DELAY HISTOGRAM ORDINATES
+// outputs used in subsequent functions include tch
+extern void convert_dist_to_histords(double *dist_from_outlet, int num_channels,
+					double chv, double rv, double dt, double tch[11])
+{
+    // declare local variables
+    double chvdt, rvdt;
+    int j;
+    
+    /*  Convert parameters to m/time step DT
+    *  with exception of XK0 which must stay in m/h
+    *                    Q0 is already in m/time step
+                     T0 is input as Ln(To) */
+
+    chvdt = chv*dt;
+    rvdt = rv*dt;
+
+    tch[1]=dist_from_outlet[1]/chvdt;
+    for(j=2;j<=num_channels;j++)
+     {
+      tch[j]=tch[1]+(dist_from_outlet[j]-dist_from_outlet[1])/rvdt;
+     }
+
+    return;
+
+}
+
+// FUNCTION TO CALCULATE THE TIME_DELAY_HISTOGRAM
+// updates values associated with model structure
+// outputs used in TOPMOD() include num_time_delay_histo_ords, time_delay_histogram,
+// num_delay, and Q
+extern void calc_time_delay_histogram(int max_time_delay_ordinates, int num_channels, double area,
```

Comment:
A common C style guide/implementation is to put all input params (things that are NOT modified by the function) first in the arg list, and all out or in/out params (things that will be modified by the function) last.  I strongly recommend this! 

Also its imperative in C codes to document assumptions on memory already being allocated vs the memory to be allocated and returned by the function.  As noted below, this function could NOT check and allocate the `time_delay_histogram`, which is valid, but it should be documented in the doc string (see above comment on doxygen style doc strings) either way.  Keeping the allocation his is OK, as long as the caller knows it was allocated in this function and can potentially deallocate it later.